internal/server/data/group_test.go

```diff
@@ -126,6 +126,41 @@ func TestListGroups(t *testing.T) {
 			}
 			assert.DeepEqual(t, actual, expected, cmpGroupShallow)
 		})
```

Comment:
There are also tests for `internal/server/handlers_test.go` however I realized that there isn't a way to add users to groups yet through the API, so those will have to wait until I add that part.